code/_onclick/hud/map_popups.dm

```diff
@@ -0,0 +1,86 @@
+/client/
+	var/list/screen_maps = list() //assoc list with all the active maps - when a screen obj is added to a map, it's put in here as well. "mapname" = list(screen objs in map)
+
+/obj/abstract/screen
+	var/assigned_map = null
+	var/list/screen_info = list()//x,x pix, y, y pix || x,y
+
+/client/proc/clear_all_popups(var/map_to_clear = null)//not really needed most of the time, as the client's screen list gets reset on relog. any of the buttons are going to get caught by garbage collection anyway. they're effectively qdel'd.
+	if(!map_to_clear)
+		return FALSE
+	for(var/obj/abstract/screen/searched in screen)
+		if(searched.assigned_map == map_to_clear)
+			qdel(searched)
+
+/client/verb/handle_popup_close(window_id as text) //when the popup closes, it calls this.
+	set hidden = 1
+	for(var/obj/abstract/screen/screenobj in screen_maps["[window_id]_map"])
+		screen -= screenobj
+		qdel(screenobj)
```

Comment:
this is technically able to be entered manually - but all that'd do is remove everything on the popup, leaving a blank black box. It's not really exploitable.